### PR TITLE
feat: add revert cases

### PR DESCRIPTION
**Pull request by keroro520**
This PR adds test cases of revert errors for https://github.com/nervosnetwork/godwoken-web3/pull/512 . It should be merged before https://github.com/nervosnetwork/godwoken-web3/pull/512 so that makes CI passed.

### CODE REVIEW

**Review by Flouse — CHANGES_REQUESTED** [review]
Please rebase develop branch when you get a moment.

**Comment by keroro520** [comment]
> Please rebase develop branch when you get a moment.

Done

**Comment by keroro520** [comment]
Wait for testnet deployment.

**Comment by Flouse** [comment]
## Related issue
- https://github.com/nervosnetwork/godwoken-web3/issues/293

**Review by Flouse — APPROVED** [review]

File: contracts/test/Revertal.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+const {isGwMainnetV1} = require('../utils/network');
+const {fetchJson} = require("ethers/lib/utils");
+
+describe("Revertal", function () {
+    // Skip for gw_mainnet_v1 network
+    if (isGwMainnetV1()) {
+        return;
+    }
+    // Skip for hardhat network
+    if (network.name === "hardhat") {
```
Comment (ShookLyngs) [inline contracts/test/Revertal.js]: Can this be added to "../utils/network" for more convenient usage search?

Comment (keroro520) [reply contracts/test/Revertal.js]: Sounds good. Like the below code?

```
function isHardhatNetwork() {
  return isNetwork(Networks.Hardhat);
}
```

Comment (ShookLyngs) [reply contracts/test/Revertal.js]: > Sounds good. Like the below code?
> 
> ```
> function isHardhatNetwork() {
>   return isNetwork(Networks.Hardhat);
> }
> ```

Yes, it should be easier for us to search for where it is used.

**Review by gpBlockchain — APPROVED** [review]

**Comment by Flouse** [comment]
Is this PR ready to be merged? @keroro520

**Comment by keroro520** [comment]
> Is this PR ready to be merged? @keroro520

> Wait for testnet deployment.

Because the testcases will be running on testnet, we have to wait for the testnet deployment before merging this PR.

**Pull request merged**